common/achievement/Readme.md

```diff
@@ -1,4 +1,192 @@
+| `metrics.ts`    | functions to register one or more metrics and to get metrics based on an action id                                                                                                                                                   |
+| `template.ts`   | functions to get all templates presorted by group or metrics and to get templates with a certain metric                                                                                                                              |
+| `types.ts`      | additional types used for the process of evaluation and creation of achievements                                                                                                                                                     |
+| `util.ts`       | additional functions for achievement processing                                                                                                                                                                                      |
+
+### 1.3. Resolvers
+
+All resolvers can be found in `common/achivement/get.ts`
+
+1.  `getAchievementById`: function to get an achievement of a user based on an achievement id
+2.  `getNextStepAchievements`: function to get a users achievements that will be displayed as a next step card on their dashboard
+3.  `getFurtherAchievements`: function to get the achievement templates, which do not yet have an associated user achievement for the corresponding user but belong to an already initialized achievement group, as inactive achievements.
+4.  `getUserAchievements`: function to get all user achievements found for a user
+
+## 2. Achievement System
+
+The achievement system is set up as follows
+
+```mermaid
+flowchart TD
+	A(actionTaken) --> B(rewardActionTaken)
+	C[Achievement Template] --> getOrCreateUserAchievement
+	D[User Id] --> getOrCreateUserAchievement
+	E[Action Event Context] --> getOrCreateUserAchievement
+	B --> getOrCreateUserAchievement
+	getOrCreateUserAchievement --> F[User Achievement]
+	F --> G(checkUserAchievement)
+	G --> isAchievementConditionMet
+	isAchievementConditionMet --> H[Evaluation Result]
+	H --> L{condition Met}
+	L -->|true| I
+	L -->|false| M(updateUserAchievement)
+	I(rewardUser) --> J[Awarded Achievement]
+	J --> K(createUserAchievement)
+
+	subgraph getOrCreateUserAchievement
+		AA(findUserAchievement) -.-> AB{User Achievement ?}
+		AB -.-> |true| AD(return User Achievement)
+		AB -.-> |false| AC(createUserAchievement)
+		AC -.-> AD
+	end
+	subgraph isAchievementConditionMet
+		BA(isAchievementConditionMet) -.-> BB(evaluateAchievement)
+		BB -.-> |return evaluation result| BC[Data Aggregation - Chapter 2.5]
+	end
+```
+
+[Go to Data Aggregation - Chapter 2.5](#data-aggregation)
+
+### 2.1. Reward Action
```

Comment:
I think we should add some more details of tracking events here. Soo that we can clearify, that each action which is stored in a metric, tracks an event.